mesonbuild/mtest.py

```diff
@@ -492,7 +492,7 @@ def emit_progress(self, harness: 'TestHarness') -> None:
             spaces=' ' * TestResult.maxlen(),
             dur=int(time.time() - self.progress_test.starttime),
             durlen=harness.duration_max_len,
-            timeout=int(self.progress_test.timeout))
+            timeout=int(self.progress_test.timeout or -1))
```

Comment:
This will replace 0 with -1, that's the desired behavior right?

Reply:
It can't be 0 at this point, timeout is either >0 or None. This is just the string that display progress, for infinite timeout it would be printed as `5/-1s`.